File: GYM.Controller.Api/DTOs/BookingDTO.cs
using GYM.Data.Entities;

namespace GYM.Controller.Api.DTOs;


public class BookingDTO
{
    public int Id { get; set; }
    public int TrainingId { get; set; }
    public int UserId { get; set; }
    public BookingStatus Status { get; set; }
    public DateTime? ExerciseTime { get; set; } 
    public DateTime? DoneAt { get; set; } 

    public List<TrainingDTO> Trainings { get; set; } = new List<TrainingDTO>();
    
    // FIX: Change this from UserDetailsDTO to UserAdminDTO to match your service layer projection map
    public List<UserAdminDTO> Users { get; set; } = new List<UserAdminDTO>();
}

File: GYM.FrontEnd/src/types/BookingDTO.ts
import type { TrainingDTO } from "./trainingDTO";
import type { UserData } from "./user";

export type BookingStatus = 'Booked'|'Working' | 'Completed'| 'Cancelled'

export interface BookingDTO {
  id?: number;                  
  trainingId: number;       
  userId: number;    
  status: BookingStatus;       
  exerciseTime: string;       
  doneAt?: string;          
  trainings: TrainingDTO[];    
  users: UserData[];
}

export interface AdminBookingDTO {
    id?: string | number;
    trainingName: string;
    status: BookingStatus;
    doneAt: string; // ISO String o Formato Fecha
    email: string;
    name: string;
    surname?: string;
}


